xesmf/frontend.py

```diff
@@ -314,9 +323,19 @@ def __init__(
         self.weights = read_weights(weights, self.n_in, self.n_out)
 
         # replace zeros by NaN in mask
-        if self.grid_out.mask is not None and self.grid_out.mask[0] is not None:
+        if default_weights is True:
             self.weights = add_nans_to_weights(self.weights)
 
+        # Warning to be removed when the default value changes in future releases
```

Comment:
I would hold off on making such a statement, let's see how often the new option is used and decide about changing defaults later

Reply:
I will remove the warning message then. After the discussion in #56 I thought that it should become the default. Again, for my purpose, that is not important so I will not try to convince you :)